src/UNL/VisitorChat/Util.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace UNL\VisitorChat;
+
+class Util
+{
+    public static function isAllowedDomain($url, array $domains)
+    {
+        if (!$parts = parse_url($url)) {
+            //it isn't a valid URL, don't allow
+            return false;
+        }
+        
+        if (!isset($parts['host'])) {
+            //it isn't a valid host, don't allow
+            return false;
+        }
+
+        $regex = "";
+        foreach ($domains as $domain) {
+            $regex .= preg_quote($domain, ".-/") . "|";
+        }
+
+        $regex = trim($regex, "|");
+        if (preg_match("/" . $regex . "$/", $parts['host'])) {
```

Comment:
The RegEx is close, but you'll want to wrap the `$regex` in parens `()` (group) to ensure the `$` (end constraint) is enforced for all of the domain combinations. Example of the problem: `/unl\.edu|nebraska\.edu$/` will match `unl.edu.phisherman.evil`. `/(unl\.edu|nebraska\.edu)$/`will not.